domaincommon/src/state.rs

```diff
@@ -46,13 +46,27 @@ impl MatchState {
 
     fn validate_has_mech(state: &MatchState, mech: &str) -> Result<()> {
         if !state.mechs.contains_key(mech) {
-            return Err(eventsourcing::Error {
+            Err(eventsourcing::Error {
                 kind: eventsourcing::Kind::CommandFailure(
                     "Command received for mech not in match".to_string(),
                 ),
-            });
+            })
+        } else {
+            Ok(())
+        }
+    }
+
+    fn validate_can_take_action(state: &MatchState, mech: &str, cmd: &MechCommand) -> Result<()> {
```

Comment:
Do we really need a result type here, or can we get away with a boolean? I only ask because returning a boolean here just makes the higher-order logic during the turn advance a little easier to read.

Reply:
However, I can see that we've kind of used results by convention, so don't feel compelled to change it.

Reply:
I think that result type works well here in place of boolean (e.g. two possible results) and would let us do something with the error message later